build error: setfl binding missing

```
error[E0560]: struct `bindings::bindings::file_operations` has no field named `setfl`
   --> /run/media/aj/Buffer/git-clones/linux-kernel-module-rust/src/chrdev.rs:228:13
    |
228 |             setfl: None,
    |             ^^^^^ `bindings::bindings::file_operations` does not have this field
    |
    = note: available fields are: `owner`, `llseek`, `read`, `write`, `read_iter` ... and 27 others

error: aborting due to previous error
```

--- a/build.rs
+++ b/build.rs
@@ -83,14 +83,10 @@ fn handle_kernel_version_cfg(bindings_path: &PathBuf) {
     if version >= kernel_version_code(5, 1, 0) {
         println!("cargo:rustc-cfg=kernel_5_1_0_or_greater")
     }
+}
 
-    let kdir = env::var("KDIR").unwrap_or(format!(
-        "/lib/modules/{}/build",
-        std::str::from_utf8(&(Command::new("uname").arg("-r").output().unwrap().stdout))
-            .unwrap()
-            .trim()
-    ));
-    let f = BufReader::new(fs::File::open(PathBuf::from(kdir).join("Module.symvers")).unwrap());
+fn handle_kernel_symbols_cfg(symvers_path: &PathBuf) {
+    let f = BufReader::new(fs::File::open(symvers_path).unwrap());
     for line in f.lines() {
         let line = line.unwrap();
         if let Some(symbol) = line.split_ascii_whitespace().nth(1) {
@@ -104,6 +100,13 @@ fn handle_kernel_version_cfg(bindings_path: &PathBuf) {
 
 fn main() {
     println!("cargo:rerun-if-env-changed=KDIR");
+    let kdir = env::var("KDIR").unwrap_or(format!(
+        "/lib/modules/{}/build",
+        std::str::from_utf8(&(Command::new("uname").arg("-r").output().unwrap().stdout))
+            .unwrap()
+            .trim()
+    ));
+
     println!("cargo:rerun-if-env-changed=CLANG");
     println!("cargo:rerun-if-changed=kernel-cflags-finder/Makefile");
     let output = Command::new("make")
@@ -153,6 +156,7 @@ fn main() {
         .expect("Couldn't write bindings!");
 
     handle_kernel_version_cfg(&out_path.join("bindings.rs"));
+    handle_kernel_symbols_cfg(&PathBuf::from(&kdir).join("Module.symvers"));
 
     let mut builder = cc::Build::new();
     println!("cargo:rerun-if-env-changed=CLANG");

--- a/build.rs
+++ b/build.rs
@@ -83,6 +83,23 @@ fn handle_kernel_version_cfg(bindings_path: &PathBuf) {
     if version >= kernel_version_code(5, 1, 0) {
         println!("cargo:rustc-cfg=kernel_5_1_0_or_greater")
     }
+
+    let kdir = env::var("KDIR").unwrap_or(format!(
+        "/lib/modules/{}/build",
+        std::str::from_utf8(&(Command::new("uname").arg("-r").output().unwrap().stdout))
+            .unwrap()
+            .trim()
+    ));
+    let f = BufReader::new(fs::File::open(PathBuf::from(kdir).join("Module.symvers")).unwrap());
+    for line in f.lines() {
+        let line = line.unwrap();
+        if let Some(symbol) = line.split_ascii_whitespace().nth(1) {
+            if symbol == "setfl" {
+                println!("cargo:rustc-cfg=kernel_aufs_setfl");
+                break;
+            }
+        }
+    }
 }
 
 fn main() {

--- a/src/chrdev.rs
+++ b/src/chrdev.rs
@@ -225,6 +225,7 @@ impl FileOperationsVtable {
             #[cfg(kernel_4_20_0_or_greater)]
             remap_file_range: None,
             sendpage: None,
+            #[cfg(kernel_aufs_setfl)]
             setfl: None,
             setlease: None,
             show_fdinfo: None,

